fairness/metrics.py

```diff
@@ -1,11 +1,206 @@
+                lst_unfairness.append(diff_quantile(data1, data2))
+            new_list.append(max(lst_unfairness))
+    return max(new_list)
+
+def unfairness_multi(y_fair_dict, x_sa_test):
+    """
+    Compute unfairness values for multiple fair output datasets and multiple sensitive attributes datasets.
+
+    Parameters:
+    y_fair_dict (dict): A dictionary where keys represent sensitive features and values are arrays
+            containing the fair predictions corresponding to each sensitive feature.
+            Each sensitive feature's fairness adjustment is performed sequentially,
+            ensuring that each feature is treated fairly relative to the previous ones.
+    x_sa_test (array-like): Sensitive attribute data.
+
+    Returns:
+    dict: A dictionary containing unfairness values for each level of fairness.
+
+    Example:
+    >>> y_fair_dict = {'Base model':np.array([19,39,65]), 'sens_var_1':np.array([22,40,50]), 'sens_var_2':np.array([28,39,42])}
+    >>> x_sa_test = np.array([['blue', 2], ['red', 9], ['green', 5]])
+    >>> unfs_dict = unfairness_multi(y_fair_dict, x_sa_test)
+    >>> print(unfs_dict)
+    {'sens_var_0': 46.0, 'sens_var_1': 28.0, 'sens_var_2': 14.0}
+    """
+    unfairness_dict = {}
+    for i, y_fair in enumerate(y_fair_dict.values()):
+        result = unfairness(y_fair, x_sa_test)
+        unfairness_dict[f'sens_var_{i}'] = result
+    return unfairness_dict
+
+
+def unfairness_multi_permutations(permut_y_fair_dict, all_combs_x_sa_test):
+    """
+    Compute unfairness values for multiple fair output datasets and multiple sensitive attribute datasets.
+
+    Parameters:
+    permut_y_fair_dict (dict): A dictionary containing permutations of fair output datasets.
+    all_combs_x_sa_test (dict): A dictionary containing combinations of columns permutations for sensitive attribute datasets.
+
+    Returns:
+    list: A list of dictionaries containing unfairness values for each permutation of fair output datasets.
+
+    Example:
+    >>> permut_y_fair_dict = {(1,2): {'Base model':np.array([19,39,65]), 'sens_var_1':np.array([22,40,50]), 'sens_var_2':np.array([28,39,42])},
+                               (2,1): {'Base model':np.array([19,39,65]), 'sens_var_2':np.array([34,39,60]), 'sens_var_1':np.array([28,39,42])}}
+    >>> all_combs_x_sa_test = {(1,2): np.array([['blue', 2], ['red', 9], ['green', 5]]),
+                               (2,1): np.array([[2, 'blue'], [9, 'red'], [5, 'green']])}
+    >>> unfs_list = unfairness_multi_permutations(permut_y_fair_dict, all_combs_x_sa_test)
+    >>> print(unfs_list)
+    [{'sens_var_0': 46.0, 'sens_var_1': 28.0, 'sens_var_2': 14.0}, 
+        {'sens_var_0': 46.0, 'sens_var_1': 26.0, 'sens_var_2': 14.0}]
+    """
+    unfs_list = []
+    for key in permut_y_fair_dict.keys():
+        unfs_list.append(unfairness_multi(
+            permut_y_fair_dict[key], np.array(all_combs_x_sa_test[key])))
+    return unfs_list
+
+
+def risk(y_true, y_predict, classif=False):
```

Comment:
this is too restrictive. As far as I can tell, these metrics are only used for the visualisation part? If so take this out of this module and work with functions/classes that have a default (for example MSE) but accept other functions

For example:


```suggestion
def estimate_risk(y_true: what_ever_type, 
                  y_pred: what_ever_type, 
                  metric:  sklearn.metrics.mean_squared_error)
                  
```